storage/repository/parse.go

```diff
@@ -43,14 +44,12 @@ func NewFileURI(path string) fyne.URI {
 // Since: 2.0
 func ParseURI(s string) (fyne.URI, error) {
 	// Extract the scheme.
-	scheme := ""
-	for i := 0; i < len(s); i++ {
-		if s[i] == ':' {
-			break
-		}
-		scheme += string(s[i])
+	colonIndex := strings.Index(s, ":")
+	if colonIndex == -1 {
```

Comment:
Should we check `== 0` here as well? Your test indicates that this already produces an error but does so further down in the chain. Thus, the question is if we should check it here when we already have the index?

This is merely a question and not a code suggestion.

Reply:
This would indicate that the scheme is blank. I can't imagine a good use case for this, but I wouldn't mind deferring to someone who has more experience with how these URI's are used. I don't think the RFC allows blank schemes at any rate. I'll throw that change in.

Reply:
It's not allowed but given the number of bugs open for crashes with invalid URIs we should probably be more defensive.